src/Stratis.Bitcoin/Controllers/NetworkController.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Controllers.Models;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities.Extensions;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+
+namespace Stratis.Bitcoin.Controllers
+{
+    /// <summary>
+    /// Provides methods that interact with the network elements of the full node.
+    /// </summary>
+    [Route("api/[controller]")]
+    public sealed class NetworkController : Controller
+    {
+        /// <summary>The connection manager.</summary>
+        private readonly IConnectionManager connectionManager;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>Network peer banning behaviour.</summary>
+        private readonly IPeerBanning peerBanning;
+
+        public NetworkController(
+            IConnectionManager connectionManager,
+            ILoggerFactory loggerFactory,
+            IPeerBanning peerBanning)
+        {
+            this.connectionManager = connectionManager;
+            this.peerBanning = peerBanning;
+        }
+
+        /// <summary>
+        /// Disconnects a connected peer.
+        /// </summary>
+        /// <param name="viewModel">The model that represents the peer to disconnect.</param>
+        /// <returns><see cref="OkResult"/></returns>
+        [Route("disconnect")]
+        [HttpPost]
+        public IActionResult DisconnectPeer([FromBody] DisconnectPeerViewModel viewModel)
+        {
+            try
+            {
+                var endpoint = viewModel.PeerAddress.ToIPEndPoint(0);
```

Comment:
I think this should be

```
    var endpoint = viewModel.PeerAddress.ToIPEndPoint(nodeSettings.Network.DefaultPort)
```

Reply:
In fact should we not actually pass the entire end point not just the address?
Currently we cannot disconnect peers that are on a different port

Reply:
Fixed 👍